pkg/stub/handler_test.go

```diff
@@ -83,9 +84,17 @@ func TestWithDist(t *testing.T) {
 func TestWithArchDist(t *testing.T) {
 	h, cfg, event := setup()
 	processCred(&h, cfg, t)
-	cfg.Spec.Architectures = []string{
-		v1.X86Architecture,
+	if len(cfg.Spec.Architectures) == 0 {
+		t.Errorf("arch not set on bootstrap")
+	}
+	testValue := runtime.GOARCH
+	if testValue == v1.AMDArchitecture {
+		testValue = v1.X86Architecture
+	}
+	if cfg.Spec.Architectures[0] != testValue {
+		t.Errorf("arch set to %s instead of %s", cfg.Spec.Architectures[0], runtime.GOARCH)
 	}
+
```

Comment:
It doesn't look like `validate` verifies that we've set version correctly. Needs to be updated.

Reply:
the setting of the version happens after the `setup()` call ... it does not need to be in `validate`

Reply:
we don't need it everwhere

Reply:
nevermind ... I'll do it everywhere

Reply:
unit test updated